src/overlays/gamestates/ovl_play/m_play.c

```diff
@@ -0,0 +1,868 @@
+        Gfx* sp190;
+        sp190 = POLY_OPA_DISP;
+        sp194 = gfxopen(sp190);
+
+        gSPDisplayList(OVERLAY_DISP++, sp194);
+
+        if (play->unk_1EE3 == 3) {
+            PlayState1938 sp60;
+            ScissorViewArg1 sp50;
+
+            initView(&sp60, gfxCtx2);
+            {
+                sp60.unk_120 = 10;
+            //! FAKE
+            label2:;
+            }
+            sp50.unk_04 = 0xF0, sp50.unk_0C = 0x140;
+            sp50.unk_00 = 0;
+            sp50.unk_08 = 0;
+            setScissorView(&sp60, &sp50);
+            showView1(&sp60, 0xF, &sp194);
+            play->unk_1EE8.unk_21C.unk_0C(&play->unk_1EE8, &sp194);
+        }
+
+        fbdemo_fade_draw(&play->unk_2128, &sp194);
+        fade_rgba8888_draw(&sp194, play->unk_220C);
+
+        gSPEndDisplayList(sp194++);
+
+        gfxclose(sp190, sp194);
+        POLY_OPA_DISP = sp194;
+    }
+
+    if (play->unk_1EE2 == 3) {
+        Gfx* sp48 = POLY_OPA_DISP;
+
+        fbdemo_draw(&fbdemo, &sp48);
+        POLY_OPA_DISP = sp48;
+        return 1;
+    }
+
+    PreRender_setup_renderbuf(&play->unk_1DC0, 0x140, 0xF0, gfxCtx2->unk_2E4, gfxCtx2->unk_008);
+    if (play->unk_1CBC.unk_00 == 2) {
+        func_800B0010_jp();
+        PreRender_ConvertFrameBuffer_fg(&play->unk_1DC0);
+        play->unk_1CBC.unk_00 = 3;
+    } else if (play->unk_1CBC.unk_00 >= 5) {
+        play->unk_1CBC.unk_00 = 0;
+    }
+
+    if (play->unk_1CBC.unk_00 == 3) {
+        Gfx* sp44 = POLY_OPA_DISP;
+
+        PreRender_loadFrameBufferCopy(&play->unk_1DC0, &sp44);
+        POLY_OPA_DISP = sp44;
+        return 1;
+    }
+
+    {
+        Lights* sp40 = Global_light_read(&play->lightCtx, gfxCtx2);
+
+        LightsN_list_check(sp40, play->lightCtx.listHead, NULL);
+        LightsN_disp(sp40, gfxCtx2);
+    }
+
+    Actor_info_draw_actor(play, &play->actorInfo);
+    Camera2_draw(play);
+    mMsg_Draw(play);
+    Debug_Display_output(play);
+
+    if ((play->unk_1CBC.unk_00 == 1) || (play->unk_1EE2 == 1)) {
+        Gfx* sp3C;
+        Gfx* sp38;
+
+        sp38 = POLY_OPA_DISP;
+        sp3C = gfxopen(sp38);
+
+        gSPDisplayList(OVERLAY_DISP++, sp3C);
+
+        play->unk_1DC0.unk_10 = gfxCtx2->unk_2E4;
+        play->unk_1DC0.unk_14 = gfxCtx2->unk_008;
+        PreRender_saveFrameBuffer(&play->unk_1DC0, &sp3C);
+
+        if (play->unk_1CBC.unk_00 == 1) {
+            play->unk_1DC0.unk_18 = gfxCtx2->unk_2E4;
+            PreRender_saveCVG(&play->unk_1DC0, &sp3C);
+            play->unk_1CBC.unk_00 = 2;
+        } else {
+            play->unk_1EE2 = 2;
+        }
+
+        gSPEndDisplayList(sp3C++);
+
+        gfxclose(sp38, sp3C);
+        POLY_OPA_DISP = sp3C;
+
+        REGADDR(SREG, 0x21) |= 1;
+        return 0;
+    }
+    return 1;
+}
+
+void draw_version(GraphicsContext* gfxCtx) {
+    Gfx* temp_s0;
+    gfxprint* printer;
+
+    OPEN_DISPS(gfxCtx);
+
+    temp_s0 = func_800BD720_jp(OVERLAY_DISP);
+    printer = (gfxprint*)alloca(sizeof(gfxprint));
```

Comment:
This isn't C++, you don't need to cast a void* to another pointer type, just assign it.
```suggestion
    printer = alloca(sizeof(gfxprint));
```